cmd/cstor-pool-mgmt/controller/replica-controller.go

```diff
@@ -0,0 +1,342 @@
+		return nil
+	}
+
+	cStorVolumeReplicaUpdated, err := c.clientset.OpenebsV1alpha1().CStorVolumeReplicas().Get(name, metav1.GetOptions{})
+	if err != nil {
+		// The CStorReplica resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("cStorReplicaUpdated '%s' in work queue no longer exists", key))
+			return nil
+		}
+
+		return err
+	}
+
+	switch operation {
+	case "add":
+		glog.Infof("added event")
+
+		err := checkValidVolumeReplica(cStorVolumeReplicaUpdated)
+		if err != nil {
+			return err
+		}
+
+		glog.Info("cstor-pool-guid:", cStorVolumeReplicaUpdated.ObjectMeta.Annotations["openebs.io/cstor-pool-guid"])
+		poolname, err := getPoolName()
+		if err != nil {
+			return err
+		}
+
+		fullvolname := string(poolname) + "/" + cStorVolumeReplicaUpdated.Spec.VolName
+		glog.Infof("fullvolname: %s", fullvolname)
+
+		err = createVolume(cStorVolumeReplicaUpdated, fullvolname)
+		if err != nil {
+			return err
+		}
+		break
+	case "update":
+		glog.Infof("updated event")
+		break
+	case "delete":
+		glog.Infof("deleted event")
+		break
+	}
+
+	return nil
+}
+
+// enqueueCStorReplica takes a CStorReplica resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorReplica.
+func (c *CStorVolumeReplicaController) enqueueCStorReplica(obj interface{}, q QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// getPoolName return the pool already created
+func getPoolName() (string, error) {
+	poolnameStr := "zpool status | grep pool:"
+	poolnamecmd := exec.Command("bash", "-c", poolnameStr)
+	stderr, err := poolnamecmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Unable to get poolname :%v ", err)
+	}
+	noisyPoolname := string(stderr)
+	poolname := strings.TrimPrefix(noisyPoolname, "  pool: ")
+	poolname = strings.TrimSpace(poolname)
+	if glog.V(4) {
+		glog.Infof("poolname : ", poolname)
+	}
+	return poolname, nil
+}
+
+// checkValidVolumeReplica checks for validity of cStor replica resource.
+func checkValidVolumeReplica(cStorVolumeReplicaUpdated *apis.CStorVolumeReplica) error {
+	if cStorVolumeReplicaUpdated.Spec.VolName == "" {
+		return fmt.Errorf("Volume name cannot be empty")
+	}
+	if cStorVolumeReplicaUpdated.Spec.Capacity == "" {
+		return fmt.Errorf("Capacity cannot be empty")
+	}
+	return nil
+}
+
+// createVolume creates cStor replica(zfs volumes).
+func createVolume(cStorVolumeReplicaUpdated *apis.CStorVolumeReplica, fullvolname string) error {
+	var createVolAttr []string
+	createVolAttr = append(createVolAttr, "create", "-s",
+		"-V", cStorVolumeReplicaUpdated.Spec.Capacity, fullvolname)
+	volCmd := exec.Command("zfs", createVolAttr...)
+	if glog.V(4) {
```

Comment:
`if ` loop not required here.
just `glog.V(4).Info ("message")` will be good enough as debug log.

Reply:
will change